pkg/kubelet/volume/reconciler/reconciler.go

```
@@ -0,0 +1,231 @@
+						mountedVolume.VolumeName,
+						mountedVolume.OuterVolumeSpecName,
+						mountedVolume.PodName,
+						mountedVolume.PodUID)
+				}
+			}
+		}
+
+		// Ensure volumes that should be attached/mounted are attached/mounted.
+		for _, volumeToMount := range rc.desiredStateOfWorld.GetVolumesToMount() {
+			volMounted, err := rc.actualStateOfWorld.PodExistsInVolume(volumeToMount.PodName, volumeToMount.VolumeName)
+			if cache.IsVolumeNotAttachedError(err) {
+				// Volume is not attached, it should be
+				if rc.controllerAttachDetachEnabled || !volumeToMount.PluginIsAttachable {
+					// Kubelet not responsible for attaching or this volume has a non-attachable volume plugin,
+					// so just add it to actualStateOfWorld without attach.
+					addVolumeNodeErr := rc.actualStateOfWorld.MarkVolumeAsAttached(
+						volumeToMount.VolumeSpec, rc.hostName)
+					if addVolumeNodeErr != nil {
+						glog.Errorf(
+							"actualStateOfWorld.MarkVolumeAsAttached failed for volume %q (spec.Name: %q) pod %q (UID: %q) controllerAttachDetachEnabled: %v with err: %v",
+							volumeToMount.VolumeName,
+							volumeToMount.VolumeSpec.Name(),
+							volumeToMount.PodName,
+							volumeToMount.Pod.UID,
+							rc.controllerAttachDetachEnabled,
+							addVolumeNodeErr)
+					}
+				} else {
+					// Volume is not attached to node, kubelet attach is enabled, volume implements an attacher,
+					// so attach it
+					volumeToAttach := operationexecutor.VolumeToAttach{
+						VolumeName: volumeToMount.VolumeName,
+						VolumeSpec: volumeToMount.VolumeSpec,
+						NodeName:   rc.hostName,
+					}
+					glog.V(12).Infof("Attempting to start AttachVolume for volume %q (spec.Name: %q)  pod %q (UID: %q)",
+						volumeToMount.VolumeName,
+						volumeToMount.VolumeSpec.Name(),
+						volumeToMount.PodName,
+						volumeToMount.Pod.UID)
+					err := rc.operationExecutor.AttachVolume(volumeToAttach, rc.actualStateOfWorld)
+					if err == nil {
```

you're swallowing another error here